QueryBuilder/Typed/DefaultQuery.cs

```diff
@@ -19,14 +20,32 @@ internal DefaultQuery(IDictionary<string, Type> aliasToTypeMapping, SelectClause
         {
         }
 
+        /// <summary>
+        /// Select a property on a type and optionally provide an alias.
+        /// </summary>
+        /// <typeparam name="TSelect">The type from which to select a property.</typeparam>
+        /// <typeparam name="TOut">The type of the property.</typeparam>
+        /// <param name="propertySelector">The property to select on the type.</param>
+        /// <param name="propertyAlias">Optional alias to map to the selected property.</param>
+        /// <param name="typeAlias">Optional alias to map to the selected type.</param>
+        /// <returns>ADT query instance with one select clause.</returns>
+        public Query<TOut> Select<TSelect, TOut>(Expression<Func<TSelect, TOut>> propertySelector, string propertyAlias = null, string typeAlias = null)
```

Comment:
I feel like either we should just make this `T` or something other than `TOut` because there's already a concept of a generic `in` or `out` generic type i.e. `Something<out T>` which could make `TOut` a tad misleading or confusing.

Reply:
Good point on the covariance / out keyword. I'm going to rename `TOut` to `TProperty` -- `T` is already used in `DefaultSelectQuery<T>` and I'd prefer to keep the naming consistent.